packages/mui-system/src/styleFunctionSx/styleFunctionSx.d.ts

```diff
@@ -48,8 +55,7 @@ export type SystemCssProperties<Theme extends object = {}> = {
 export type SystemStyleObject<Theme extends object = {}> =
   | SystemCssProperties<Theme>
   | CSSPseudoSelectorProps<Theme>
-  | CSSSelectorObject<Theme>
-  | { [cssVariable: string]: string | number }
```

Comment:
`[cssVariable: string]` is basically the same as `[cssSelector: string]` defined in CSSSelectorObject, so whichever will be mapped first, will invalidate the other, hence the CSS variables cannot be defined in the same object as nested selectors (see the tests)